### PR TITLE
cripto list finished

### PR DESCRIPTION
**Pull request by LuisGerezM**
Entrega de 1er feature. Quiero aclarar que utilizo un dispositivo físico, android, para poder levantar la app.
1ra imagen:  lazy loading para las imagenes:
Screenshot_20230322-155104

2da imagen: se muestra la vista acorde a los observado en Figma.
Screenshot_20230322-155120

### CODE REVIEW

File: src/App.styled.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import styled from 'styled-components/native';
+
+const Container = styled.SafeAreaView`
```
Comment (FedericoLeiva12) [inline src/App.styled.ts:3]: Are you using this style?

File: src/components/Buttons/ButtonToNavigate/ButtonToNavigate.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}: ButtonToNav) => {
+  const navigation = useNavigation();
+
+  const handlerAddPressed = (): void => {
+    // navigation.navigate(to);
+  };
+
+  return (
+    <NavigationButton onPress={handlerAddPressed}>
+      {children || null}
```
Comment (FedericoLeiva12) [inline src/components/Buttons/ButtonToNavigate/ButtonToNavigate.tsx:28]: Is this validation needed?

File: src/bd-mock/criptoData.mock.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    symbol: 'ETH',
+    price_usd: 146.83,
+    percent: 1.82,
+    icon: criptoSchema.criptosImg.ethereum,
+  },
+  {
+    name: 'XRP',
+    symbol: 'XRP',
+    price_usd: 0.22,
+    // percent_change_usd_last_24_hours -> percent
```
Comment (FedericoLeiva12) [inline src/bd-mock/criptoData.mock.ts:25]: delete comments

File: src/bd-mock/criptoData.mock.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import {Cripto} from 'src/models/cripto.models';
+import {criptoSchema} from 'src/schema/cripto.schema';
+
+// export const mockCriptosData: Cripto[] = [];
```
Comment (FedericoLeiva12) [inline src/bd-mock/criptoData.mock.ts:4]: delete comments

File: src/components/Buttons/ButtonToNavigate/ButtonToNavigate.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  fontSize = 'text',
+  fontWeight = 'bold',
+  color = 'blueGray',
+  text = '',
+  to,
+}: ButtonToNav) => {
+  const navigation = useNavigation();
+
+  const handlerAddPressed = (): void => {
+    // navigation.navigate(to);
```
Comment (FedericoLeiva12) [inline src/components/Buttons/ButtonToNavigate/ButtonToNavigate.tsx:23]: delete comments

File: src/components/Images/ImageComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+import {ImageIcon} from 'src/styled-components/globalStyles/GlobalStyles.styled';
+
+interface Image {
```
Comment (FedericoLeiva12) [inline src/components/Images/ImageComponent.tsx:5]: Why are all the props optional and handleLoadingImg not? Is this correct? Can you use this component without sending a src for example?

File: src/components/Spinner/SpinnerLoader.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+export const SpinnerLoader = ({color}: SpinnerProps): JSX.Element => {
+  return (
+    <WrapSpinnerLoad>
+      <ActivityIndicator size="large" color={color} />
+    </WrapSpinnerLoad>
+  );
+};
+
+const WrapSpinnerLoad = styled.View`
```
Comment (FedericoLeiva12) [inline src/components/Spinner/SpinnerLoader.tsx:17]: Move the styles to a `styles.ts` file

File: src/models/cripto.models.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+export interface Cripto {
```
Comment (FedericoLeiva12) [inline src/models/cripto.models.ts:1]: Crypto*

File: src/pages/Home/components/CriptoCard/CriptoCard.tsx [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+import {TextComponent} from 'src/styled-components/globalStyles/GlobalStyles.styled';
+import {defaultTheme} from 'src/styled-components/theme/theme.styled';
+import {
+  Cripto,
+  WrapNameAndIconCripto,
+  WrapPriceAndPercentCripto,
+  WrapStatusPercent,
+} from './styled-components/CriptoCard.styled';
+
+interface CriptoC {
```
Comment (FedericoLeiva12) [inline src/pages/Home/components/CriptoCard/CriptoCard.tsx:15]: When you're defining an interface to use for props, you have to use a name like `ComponentNameProps`, in this case for example `CriptoCardProps`.

Comment (FedericoLeiva12) [reply src/pages/Home/components/CriptoCard/CriptoCard.tsx:15]: Also, change Cripto to Crypto

File: src/pages/Home/components/CriptoCard/styled-components/CriptoCard.styled.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import styled from 'styled-components/native';
+
+const Cripto = styled.View`
```
Comment (FedericoLeiva12) [inline src/pages/Home/components/CriptoCard/styled-components/CriptoCard.styled.ts:3]: Crypto* there are many places here where you're using "cripto".

File: src/pages/Home/hooks/useCriptoCard.ts [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import {useSpinnerLoading} from 'src/hooks/useSpinnerLoading';
+
+export const useCriptoCard = () => {
+  const {loadImg, handleLoadingImg} = useSpinnerLoading();
+
+  const sourceByPercent = (percent: number) =>
+    percent > 0
+      ? require('../../../assets/icons/arrowRigthTop-24.png')
```
Comment (FedericoLeiva12) [inline src/pages/Home/hooks/useCriptoCard.ts:8]: Right*

File: src/models/routes.models.ts [hunk 12]
```diff
@@ -0,0 +1,8 @@
+export interface RoutesApp {
+  [key: string]: string;
+}
+
+export const routes: RoutesApp = {
+  HOME: 'Home',
+  ADDCRIPTOCURRENCY: 'add-criptocurrency',
```
Comment (FedericoLeiva12) [inline src/models/routes.models.ts:7]: You should use Screaming Snake Case (ADD_CRYPTO_CURRENCY) here.

File: src/pages/Home/hooks/useCriptoCard.ts [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+      : require('../../../assets/icons/arrayLeftBottom-24.png');
+
+  const convertNegativeNum = (num: number): number =>
```
Comment (FedericoLeiva12) [inline src/pages/Home/hooks/useCriptoCard.ts:11]: It can be done using `-Math.abs(num)`

File: src/schema/cripto.schema.ts [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
...i2MTbNS37QpnXLJVK9z5tibzIE6xXivY4rPVDLJdV1QUnL8NaB40KZsf6rCDxOHrSQJkB2gXzrNGDrbUDPE6PbF3/UttfK8vzXSQEKrzrAmL13dXUdUR5fdRBDZOhRb3sgcXcC62cJZkYPi8ITA8R1M8TB8RWgl++UzatcmKZckD3VQRwe3wDqvE82K6Z/KFcE1p8Uh8cXgAY7gJe+K58VF2crlwRDZfFQpOGojGB7eqtd7iqdE1Ojck1guk48GGlAKifx3jXA5q/JPW2d6M8q11SdYNXrAkDdVWjeaAN2/GroOBcXdlwcNTUrVxU/kUmeIS3d01S6gfb5bu02Pda2bUr1VyiXBabHxIMaK2S1+F3jgh0Y9xlJaFKTwfS6fFgBIBwPTy9Wnnm28kH2DXjioASA8LYMblW+CFBnwFBJHJbhnMtLGEX2IUf5JET1F8DQbvnwcg4Q6z6UixcqHxWPDbE+Ih5iXgFifQRMX1Q+C6x/LR5kfgF6QPkuRKoOTA3iYeYOIGpOfadFrYRV9eeDqVs+1LwARF1oqD9PZUlg+hCYXpMPN+um3YjoUpVFDS0+e0M+5Iya9QBMcZbKslDSn4bRFfGws2bWhxDpa1UeBFOYH1dYOvSsmG2WhfkqT4p7ItYHxMP33/1gmqvyqOryD4cnXl03677M3/OMJPvE4MwjvlemLjTQh6Xbz6Vxokb5RvHETA2ZG+cZlxFro+8XXdHoujmeV1ycmRHmWgiRngNDPeKN5Zxpt52clm4ff5aCMBn5RnPFFNnLvHS7eKV4lwcXF+R6+oN1H5jusgv0pNvDWyFS74qv+y6/q9WMOzhHYGiJ2MFPWZTdjgLWa8Ubt+amZjs+Jp13ZoWoeLUzhzmY8ex1qAFM13u3dtlvkPQK55bMmlGAE+k50nnmVojqpoPpkfi0La+mIOhhZ87nCVLKHtUWXwKMfio+eVQaEnOCD8PQyvjJslVNCG3msOzTy9A51sviU9jloNlnD/GON1r6sis06HjZYX9w8RNgusceaWtfa1Sj6ZLD8Wcbvdy+uARR/ZV2eia0R2Yvd4XLUKKbwLQQTI/HT3asn4ehnfH91LFrlezX8fdoB4zeUP3Z+HcWwtCNiApTxU87DQoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKUuOp/wMmNkn83ycv0gAAAABJRU5ErkJggg==',
```
Comment (FedericoLeiva12) [inline src/schema/cripto.schema.ts:12]: Why not to use image files?

File: src/styled-components/theme/theme.styled.ts [hunk 15]
```diff
@@ -0,0 +1,40 @@
+import {DefaultTheme} from 'styled-components/native';
+
+declare module 'styled-components/native' {
+  export interface DefaultTheme {
+    colors: {
+      [key: string]: string;
```
Comment (FedericoLeiva12) [inline src/styled-components/theme/theme.styled.ts:6]: This type isn't goint to allow you to use suggestions to select a color, you can use `typeof defaultTheme`

**Review by FedericoLeiva12 — CHANGES_REQUESTED** [review]

File: src/App.styled.ts (hunk 1 above)
Comment (LuisGerezM) [reply src/App.styled.ts:3]: Yes, I'm  use this style in Home.tsx.

File: src/components/Buttons/ButtonToNavigate/ButtonToNavigate.tsx (hunk 2 above)
Comment (LuisGerezM) [reply src/components/Buttons/ButtonToNavigate/ButtonToNavigate.tsx:28]: no, it isn't needed. I deleted it.

File: src/bd-mock/criptoData.mock.ts (hunk 3 above)
Comment (LuisGerezM) [reply src/bd-mock/criptoData.mock.ts:25]: Okey, sorry, I needed to remember for next feature, and in this new feature I delete. But delete now.

File: src/pages/Home/hooks/useCriptoCard.ts (hunk 13 above)
Comment (LuisGerezM) [reply src/pages/Home/hooks/useCriptoCard.ts:11]: Yes, I'm accord. Thanks.

File: src/schema/cripto.schema.ts (hunk 14 above)
Comment (LuisGerezM) [reply src/schema/cripto.schema.ts:12]: I don't now. Wanted to try another file type. I'm new in react-native, and wanted to check another shape. I modify it?

File: src/styled-components/theme/theme.styled.ts (hunk 15 above)
Comment (LuisGerezM) [reply src/styled-components/theme/theme.styled.ts:6]: sorry, I understand but I don't know use it. 

```
declare module 'styled-components/native' {
  export interface DefaultTheme {
    colors: {
      black: string;
      blueGray: string;
      darkBlue: string;
      green: string;
      grey: string;
      navyBlue: string;
      red: string;
      skyBlue: string;
      white: string;
      yellow: string;
    };
    fontSizes: {
      title: string;
      subTitle: string;
      text: string;
    };
    fontWeights: {
      heavy: number;
      bold: number;
      light: number;
    };
  }
}
```

It is good?

File: src/models/routes.models.ts (hunk 12 above)
Comment (LuisGerezM) [reply src/models/routes.models.ts:7]: Okey

File: src/pages/Home/hooks/useCriptoCard.ts (hunk 11 above)
Comment (LuisGerezM) [reply src/pages/Home/hooks/useCriptoCard.ts:8]: ups, thanks.

File: src/components/Buttons/ButtonToNavigate/ButtonToNavigate.tsx (hunk 5 above)
Comment (LuisGerezM) [reply src/components/Buttons/ButtonToNavigate/ButtonToNavigate.tsx:23]: Okey.

File: src/bd-mock/criptoData.mock.ts (hunk 4 above)
Comment (LuisGerezM) [reply src/bd-mock/criptoData.mock.ts:4]: Okey.

File: src/App.styled.ts (hunk 1 above)
Comment (javi10823) [reply src/App.styled.ts:3]: Every screen must have their own styles file.

File: src/components/Buttons/ButtonToNavigate/ButtonToNavigate.tsx (hunk 2 above)
Comment (javi10823) [reply src/components/Buttons/ButtonToNavigate/ButtonToNavigate.tsx:28]: But if children is undefined, it still works without the validation

File: src/styled-components/theme/theme.styled.ts (hunk 15 above)
Comment (javi10823) [reply src/styled-components/theme/theme.styled.ts:6]: You can keep using this, not problem.

File: src/schema/cripto.schema.ts (hunk 14 above)
Comment (LuisGerezM) [reply src/schema/cripto.schema.ts:12]: in feature 2 i'm using images of api.

File: src/App.styled.ts (hunk 1 above)
Comment (LuisGerezM) [reply src/App.styled.ts:3]: @javi10823 Yes I'm accord, but if after I need to use it in other components? I thought it's reusable

Comment (LuisGerezM) [reply src/App.styled.ts:3]: I moved it to home

**Pull request merged**